projects/components/src/form/number-with-unit-input/number-with-unit-form-input.component.html

```diff
@@ -14,7 +14,7 @@
         [maxlength]="maxlength"
         [showAsterisk]="showAsterisk"
         [label]="label"
-        [description]="description"
+        [description]="formControl.valid && description"
```

Comment:
Don't change the description. The input automatically hides/shows the description/error.